src/devices/src/virtio/block/device.rs

```diff
@@ -258,15 +258,40 @@ impl Block {
                             break;
                         }
                     }
+
                     let status = match request.execute(&mut self.disk, mem) {
                         Ok(l) => {
-                            len = l;
-                            VIRTIO_BLK_S_OK
+                            // Account for the status byte as well.
+                            if let Some(l) = l.checked_add(1) {
+                                len = l;
+                                VIRTIO_BLK_S_OK
+                            } else {
+                                len = 1;
```

Comment:
I don't think this fixes the issue :(. If we're on the `Ok(l)` path, it means `l` bytes were transferred. The problem is we shouldn't actually get to this point (Request::parse() should fail before we get to execute the command and basically, no bytes won't be transferred), but this is somehow another issue. For now, I would change this to `len = l` or use `expect()`.

Reply:
With current code, a faulty driver will get status error byte which is correct from a protocol point of view right?

We don’t want to add process fatal error/panic conditions on code paths controlled by the guest.